### DISCUSSION TITLE
Server Update

### DISCUSSION THREAD

**Discussion by wiskone** [dani-garcia/vaultwarden#2042] [Q&A]
Hello!

I´ve installed vaultwarden yesterday to my raspberry p3 like the following https://www.heise.de/ratgeber/Anleitung-Raspberry-Pi-als-Passwort-Server-einrichten-6005925.html manual and it works fine, but...
![vw1](https://user-images.githubusercontent.com/92721779/137709549-571e6098-c78c-409d-9380-dd66ccc89c9d.PNG)
...the Version is not the latest!
I use the following commands from the manual https://github.com/dani-garcia/vaultwarden/wiki/Updating-the-vaultwarden-image ...
![vw2](https://user-images.githubusercontent.com/92721779/137710019-ec16fd7b-94bd-41e8-a278-6b46d0a48c15.PNG)
...but nothin happens!
Can anybody help?

**Comment by redpixy** [comment]
what's your compose file config?

**Reply by wiskone** [reply]
Hello redpixy,

`
---
services:
  traefik:
    command: --log.level=error
    image: traefik:2.3
    container_name: traefik
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
      - ./config-files:/etc/traefik
    networks:
      - traefik_reverse_proxy
      - default
    ports:
      - "80:80"
      - "443:443"
    restart: unless-stopped

  bitwarden:
    image: vaultwarden/server:1.22.1
    container_name: vaultwarden
    restart: unless-stopped
    volumes:
      - /var/docker/bitwarden:/data
    labels:
      - "traefik.enable=true"
      - "traefik.http.middlewares.redirect-https.redirectScheme.scheme=https"
      - "traefik.http.middlewares.redirect-https.redirectScheme.permanent=true"
      - "traefik.http.routers.bitwarden-ui-https.rule=Host(`${BITWARDEN_URL}`)"
      - "traefik.http.routers.bitwarden-ui-https.entrypoints=web-secure"
      - "traefik.http.routers.bitwarden-ui-https.tls=true"
      - "traefik.http.routers.bitwarden-ui-https.service=bitwarden-ui"
      - "traefik.http.routers.bitwarden-ui-http.rule=Host(`${BITWARDEN_URL}`)"
      - "traefik.http.routers.bitwarden-ui-http.entrypoints=web"
      - "traefik.http.routers.bitwarden-ui-http.middlewares=redirect-https"
      - "traefik.http.routers.bitwarden-ui-http.service=bitwarden-ui"
      - "traefik.http.services.bitwarden-ui.loadbalancer.server.port=80"
      - "traefik.http.routers.bitwarden-admin.rule=Host(`${BITWARDEN_URL}`) && PathPrefix(`/admin`)"
      - "traefik.http.routers.bitwarden-admin.entrypoints=web-secure"
      - "traefik.http.routers.bitwarden-admin.tls=true"
      - "traefik.http.routers.bitwarden-admin.service=admin-panel"
      - "traefik.http.services.admin-panel.loadbalancer.server.port=80"
      - "traefik.http.routers.bitwarden-admin.tls.certresolver=default"
      - "traefik.http.middlewares.admin-auth.basicauth.users=user:password" #'user:password' ersetzen.
      - "traefik.http.routers.bitwarden-admin.middlewares=admin-auth"
      - "traefik.http.routers.bitwarden-ui-https.middlewares=secHeaders@file"
      - "traefik.http.routers.bitwarden-websocket-https.rule=Host(`${BITWARDEN_URL}`) && Path(`/notifications/hub`)"
      - "traefik.http.routers.bitwarden-websocket-https.entrypoints=web-secure"
      - "traefik.http.routers.bitwarden-websocket-https.tls=true"
      - "traefik.http.routers.bitwarden-websocket-https.service=bitwarden-websocket"
      - "traefik.http.routers.bitwarden-websocket-http.rule=Host(`${BITWARDEN_URL}`) && Path(`/notifications/hub`)"
      - "traefik.http.routers.bitwarden-websocket-http.entrypoints=web"
      - "traefik.http.routers.bitwarden-websocket-http.middlewares=redirect-https"
      - "traefik.http.routers.bitwarden-websocket-http.service=bitwarden-websocket"
      - "traefik.http.services.bitwarden-websocket.loadbalancer.server.port=3012"
    environment:
      - "ADMIN_TOKEN=${ADMIN_TOKEN}"
      - "WEBSOCKET_ENABLED=${WEBSOCKET_ENABLED}"
    networks:
  traefik_reverse_proxy:
    external:
      name: traefik_reverse_proxy
  default:
    driver: bridge

`


I change
    image: vaultwarden/server:1.22.1
to
    image: vaultwarden/server:1.22.2

and the command "docker-compose pull" seems to be doing an update, but it doesn't!

**Comment by redpixy** [comment]
try using 'vaultwarden/server:latest' instead

**Comment by BlackDex** [comment]
You need to reconfigure docker-compose.
To do that you need to run `docker-compose up -d` That will check for differences and apply then.
Running `docker-compose pull` will download the newer version of the compose file, but a `stop/start` doesn't reconfigure it and will keep using the previous image.

Also, `latest` would be the best if you want to have the latest tagged releases. You can use `testing` if you want to use main branch based images.

**Reply by wiskone** [reply]
It works;) Thank you for you Help redpixy and BlackDex
For my personal Doku
- Updating Vaultwarden Server when using docker-compose -
cd /srv/raspi-pwm
sudo nano docker-compose.yaml
and change image: vaultwarden/server:1.22.1 TO image: vaultwarden/server:latest
docker-compose stop
docker-compose pull
docker-compose start
docker-compose up -d
TATAAAAA
![vw3](https://user-images.githubusercontent.com/92721779/137731092-e245796a-bf36-487d-a0a6-b003e0e7e172.PNG)
